File: backend/src/main/java/ibee/webapp/todo_app/core/dto/person/skills/hard/PersonProfessionQualificationDto.java
package ibee.webapp.todo_app.core.dto.person.skills.hard;

import java.time.LocalDate;

import ibee.webapp.todo_app.core.dto.EducationInstitutionDto;
import ibee.webapp.todo_app.core.dto.person.referenceIds.skill.hard.PersonProfessionQualificationDtoId;
import ibee.webapp.todo_app.core.dto.skills.hard.ProfessionQualificationDto;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PastOrPresent;

public record PersonProfessionQualificationDto(

    
    PersonProfessionQualificationDtoId id,

    @NotBlank
    String name,

    @NotNull
    EducationInstitutionDto educationInstitution,

    @NotNull
    ProfessionQualificationDto professionQualification,

    @NotNull
    @PastOrPresent
     LocalDate startDate,

    @PastOrPresent
     LocalDate endDate,

    String certificateNumber
) {

}

File: frontend/src/app/types/dto/person/related/skill/hard/person-profession-qualification.dto.ts
import { ProfessionQualificationDto } from '../../../../skill/hard/profession-qualification.dto';
import { EducationInstitutionDto } from '../../../../education-institution.dto';
import { PersonPhoneNumberDtoId } from '../../reference/contact/person-phone-number-dto-id';


export interface PersonProfessionQualificationDto {

    id?: PersonPhoneNumberDtoId;

    professionQualification: ProfessionQualificationDto;

    educationInstitution: EducationInstitutionDto;

    startDate: Date;

    endDate?: Date;

    certificateNumber: string;

    name: string;

    

}